src/Types/Output/InterfaceType.hack

```diff
@@ -4,9 +4,15 @@ use namespace HH\Lib\{Dict, Vec};
 use namespace Slack\GraphQL;
 
 abstract class InterfaceType extends CompositeType {
+    abstract const keyset<classname<GraphQL\Types\ObjectType>> POSSIBLE_TYPES;
 
     <<__Override>>
     final public function getKind(): GraphQL\Introspection\__TypeKind {
         return GraphQL\Introspection\__TypeKind::INTERFACE;
     }
+
+    <<__Override>>
+    final public function getPossibleTypes(): vec<GraphQL\Introspection\__Type> {
```

Comment:
We could probably make this (and similar introspection methods) return `INullableType` to make it impossible to do the wrong thing.

Alternatively, maybe this could return `vec<ObjectType>` (for maximum type-safety + if this is reused from validation or elsewhere, that will probably be more useful). This is of course the wrong thing for introspection, but we can take care of that by doing a `Vec\map` inside `TNullableType::getIntrospectionPossibleTypes`, which has the bonus advantage that all "confusing introspection stuff" stays in one place -- we can't prevent TNullableType from being confusing but we can prevent everything else :)